src/protocols/types.rs

```diff
@@ -164,22 +171,39 @@ impl TimeoutSettings {
             }
         }
 
-        Ok(Self { read, write })
+        Ok(Self {
+            read,
+            write,
+            retries: retries.unwrap_or(0),
+        })
     }
 
     /// Get the read timeout.
     pub const fn get_read(&self) -> Option<Duration> { self.read }
 
     /// Get the write timeout.
     pub const fn get_write(&self) -> Option<Duration> { self.write }
+
+    /// Get number of retries
+    pub const fn get_retries(&self) -> usize { self.retries }
+
+    /// Get the number of retries if there are timeout settings else fall back
+    /// to the default
+    pub fn get_retries_or_default(timeout_settings: &Option<TimeoutSettings>) -> usize {
```

Comment:
This maybe shouldn't be in `impl TimeoutSettings` but you cannot implement functions on external types (`Optional`) so I put it here for organization. (Also probably doesn't need to be pub). Let me know what you think.

Reply:
I think thats fine for now.